docs/nddata/subclassing.rst

```diff
@@ -1,100 +1,488 @@
+    >>> ndd1.add(ndd2).mask is None  # it will be None
+    True
+
+    >>> # But giving other values is still possible:
+    >>> ndd1.add(ndd2, handle_mask=np.logical_or).mask
+    True
+
+    >>> ndd1.add(ndd2, handle_mask="ff").mask
+    False
+
+The parameter controlling how properties are handled are all keyword-only
+so using the ``*args, **kwargs`` approach allows only to alter one default
+without needing to care about the positional order of arguments. But using
+``def _arithmetic(self, *args, handle_mask=None, **kwargs)`` doesn't work
+for python 2.
+
+
+Arithmetic with an additional property
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This also requires overriding the ``_arithmetic`` method. Suppose we have a
+``flags`` attribute again::
+
+    >>> from copy import deepcopy
+    >>> import numpy as np
+
+    >>> class NDDataWithFlags(NDDataRef):
+    ...     def __init__(self, *args, **kwargs):
+    ...         # Remove flags attribute if given and pass it to the setter.
+    ...         self.flags = kwargs.pop('flags') if 'flags' in kwargs else None
+    ...         super(NDDataWithFlags, self).__init__(*args, **kwargs)
+    ...
+    ...     @property
+    ...     def flags(self):
+    ...         return self._flags
+    ...
+    ...     @flags.setter
+    ...     def flags(self, value):
+    ...         self._flags = value
+    ...
+    ...     def _arithmetic(self, operation, operand, *args, **kwargs):
+    ...         # take all args and kwargs to allow arithmetic on the other properties
+    ...         # to work like before.
+    ...
+    ...         # do the arithmetics on the flags (pop the relevant kwargs, if any!!!)
+    ...         if self.flags is not None and operand.flags is not None:
+    ...             result_flags = np.logical_or(self.flags, operand.flags)
+    ...             # np.logical_or is just a suggestion you can do what you want
+    ...         else:
+    ...             if self.flags is not None:
+    ...                 result_flags = deepcopy(self.flags)
+    ...             else:
+    ...                 result_flags = deepcopy(operand.flags)
+    ...
+    ...         # Let the superclass do all the other attributes note that
+    ...         # this returns the result and a dictionary containing other attributes
+    ...         result, kwargs = super(NDDataWithFlags, self)._arithmetic(operation, operand, *args, **kwargs)
+    ...         # The arguments for creating a new instance are saved in kwargs
+    ...         # so we need to add another keyword "flags" and add the processed flags
+    ...         kwargs['flags'] = result_flags
+    ...         return result, kwargs # these must be returned
+
+    >>> ndd1 = NDDataWithFlags([1,2,3], flags=np.array([1,0,1], dtype=bool))
+    >>> ndd2 = NDDataWithFlags([1,2,3], flags=np.array([0,0,1], dtype=bool))
+    >>> ndd3 = ndd1.add(ndd2)
+    >>> ndd3.flags
+    array([ True, False,  True], dtype=bool)
+
+Another arithmetic operation
```

Comment:
I'd move this up to the first example of subclassing `NDDataRef`. This ability is pretty nice! Also add the word "Add" to the front of the title